src/Workspaces/Remote/Core/Services/SolutionCreator.cs

```diff
@@ -497,5 +507,49 @@ private Project AddDocument(Project project, DocumentInfo documentInfo)
         {
             return project.Solution.AddDocument(documentInfo).GetProject(project.Id);
         }
+
+        private CompilationOptions FixUpCompilationOptions(ProjectInfo.ProjectAttributes info, CompilationOptions compilationOptions)
+        {
+            return compilationOptions.WithXmlReferenceResolver(new XmlFileResolver(GetXmlResolverBasePath(info.FilePath)))
+                                     .WithStrongNameProvider(new DesktopStrongNameProvider(GetStrongNameKeyPaths(info)));
+        }
+
+        private static string GetXmlResolverBasePath(string filePath)
+        {
+            if (!PathUtilities.IsAbsolute(filePath))
```

Comment:
I see. That's very odd. Seems like the path should have some meaning. If it doesn't you need to wrap ```Path.GetDirectoryName``` in try-catch-all in case the path contains invalid path characters. Or you can use FileUtilities.GetDirectoryName that doesn't throw.